types/src/validator_signer.rs

```diff
@@ -12,25 +15,22 @@ use std::convert::TryFrom;
 /// signing, respectively.
 #[derive(Debug)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Clone))]
-pub struct ValidatorSigner<PrivateKey: SigningKey> {
+pub struct ValidatorSigner {
     author: AccountAddress,
-    public_key: PrivateKey::VerifyingKeyMaterial,
-    private_key: PrivateKey,
+    private_key: Ed25519PrivateKey,
 }
 
-impl<PrivateKey: SigningKey> ValidatorSigner<PrivateKey> {
-    pub fn new(author: AccountAddress, private_key: PrivateKey) -> Self {
-        let public_key: PrivateKey::VerifyingKeyMaterial = private_key.public_key();
+impl ValidatorSigner {
+    pub fn new(author: AccountAddress, private_key: Ed25519PrivateKey) -> Self {
         ValidatorSigner {
             author,
-            public_key,
             private_key,
         }
     }
 
     /// Constructs a signature for `message` using `private_key`.
-    pub fn sign_message(&self, message: HashValue) -> Result<PrivateKey::SignatureMaterial, Error> {
```

Comment:
curious why this is a Result before

Reply:
This makes no sense to me either, since the underlying type always returns the signature, so it made for an unnecessary error handling.